### ISSUE TITLE
no-unused-state with GDSFP and airbnb plugin causes "Cannot read properties of undefined (reading 'variables')"

### ISSUE THREAD

**Issue by mattparlane** [yannickcr/eslint-plugin-react#3240]
I have come across this error after a recent upgrade, and it looks like it may be related to https://github.com/yannickcr/eslint-plugin-react/commit/9b227aa8c8d2ffacd508ff2b281f4d7852409464. This only happens with the `airbnb` plugin enabled, which my company is using. If you comment that line out from the `.eslintrc.js` file, it works fine.

Apologies if this is not the appropriate place for this, I can take it to Airbnb if it's a better fit.

Minimal `.eslintrc.js`:

```
module.exports = {
  parser: '@babel/eslint-parser',
  extends: 'airbnb',
  parserOptions: {
    requireConfigFile: false,
  },
};
```

Minimal `test.js`:
```
class Thing extends Component {
  static getDerivedStateFromProps = (nextProps, prevState) => {
  };
};
```

```
# ./node_modules/eslint/bin/eslint.js test.js
Oops! Something went wrong! :(

ESLint: 7.32.0

TypeError: Cannot read properties of undefined (reading 'variables')
Occurred while linting /root/test.js:2
    at PropertyDefinition, ClassProperty (/root/node_modules/eslint-plugin-react/lib/rules/no-unused-state.js:373:20)
    at /app/frontend/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/app/frontend/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/app/frontend/node_modules/eslint/lib/linter/node-event-generator.js:293:26)
    at NodeEventGenerator.applySelectors (/app/frontend/node_modules/eslint/lib/linter/node-event-generator.js:322:22)
    at NodeEventGenerator.enterNode (/app/frontend/node_modules/eslint/lib/linter/node-event-generator.js:336:14)
    at CodePathAnalyzer.enterNode (/app/frontend/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:711:23)
    at /app/frontend/node_modules/eslint/lib/linter/linter.js:960:32
    at Array.forEach (<anonymous>)
```

**Comment by ljharb** [comment]
Duplicate of #3236, fixed in the next release.

**Comment by mattparlane** [comment]
Thanks! Sorry, I must have missed the other issue.

**Comment by ljharb** [comment]
No worries; I’ll try to cut a release tomorrow.

**Comment by WillyLiaoWH** [comment]
Still encountered the issue in 7.29.4
The error in #3236 is about `node.value.params` is undefined. And this issue is caused by `scope` is undefined in the line below. (reading variables)
https://github.com/yannickcr/eslint-plugin-react/blob/5bc571db4743527d368167738f359dc6168f9b02/lib/rules/no-unused-state.js#L374

**Comment by ljharb** [comment]
@cbedwinotalvaro the commit that fixed this issue is in v7.29.4. https://github.com/yannickcr/eslint-plugin-react/commit/4876d549158df538520026170df3b8c933fa3603, however, was only merged 2 weeks ago and has not yet been released.

There's no timeline for the next release (as always). I hope I'll have time soon.
